Unix/deprecated/httpclientcxx/httpclientcxx.cpp

```diff
@@ -726,6 +731,22 @@ Result HttpClient::Connect(
     return httpclient::OKAY;
 }
 
+void HttpClient::Cancel( void )
+{
+    pthread_mutex_lock(&s_deleteSemaphoreLock);
+
+    NotifyItem* item = new NotifyItem(_rep);
+
+    if (!_rep->_th->PostItem(item))
+    {
+        LOGE2((ZT("HttpClient::Connect - PostItem failed")));
```

Comment:
you did not 
    pthread_mutex_unlock(&s_deleteSemaphoreLock); 
in failure case. It Does the PostItem always delete the NotifyItem even in failure case, if not then you need to delete also,